java/jdbc/src/main/java/io/vitess/jdbc/VitessJDBCUrl.java

```diff
@@ -158,11 +158,22 @@ public VitessJDBCUrl(String url, Properties info) throws SQLException {
             throw new SQLException(Constants.SQLExceptionMessages.MALFORMED_URL);
         }
 
-        this.keyspace = StringUtils.isNullOrEmptyWithoutWS(m.group(8)) ? null : m.group(8);
+        this.keyspace = StringUtils.isNullOrEmptyWithoutWS(m.group(8)) ? info.getProperty(Constants.Property.KEYSPACE) : m.group(8);
         this.catalog =
             StringUtils.isNullOrEmptyWithoutWS(m.group(10)) ? this.keyspace : m.group(10);
         this.hostInfos = getURLHostInfos(postUrl);
         this.url = url;
+
+        // For Backward Compatibity of tablet type.
+        String oldTabletType = info.getProperty(Constants.Property.OLD_TABLET_TYPE);
```

Comment:
Since we're doing a major version bump I wonder if we should just remove the old tablet type